File: exercism/ruby/space-age/space_age.rb
class SpaceAge

  SECONDS_PER_EARTH_YEAR = 31557600
  FACTORS =
    {
      mercury: 0.2408467,
      venus: 0.61519726,
      mars: 1.8808158,
      jupiter: 11.862615,
      saturn: 29.447498,
      uranus: 84.016846,
      neptune: 164.79132
    }

  def initialize(seconds)
    @seconds = seconds.to_f
  end

  def on_earth
    @seconds / SECONDS_PER_EARTH_YEAR
  end

  def convert(planet)
    @seconds / (FACTORS[planet] * SECONDS_PER_EARTH_YEAR)
  end

  def on_mercury
    convert(:mercury)
  end

  def on_venus
    convert(:venus)
  end

  def on_mars
    convert(:mars)
  end

  def on_jupiter
    convert(:jupiter)
  end

  def on_saturn
    convert(:saturn)
  end

  def on_uranus
    convert(:uranus)
  end

  def on_neptune
    convert(:neptune)
  end
end


module BookKeeping
  VERSION = 1
end

File: exercism/javascript/space-age/space-age.js
function Age(seconds) {
  var secondsPerMinute = 60;
  var minutesPerHour = 60;
  var hoursPerDay = 24;
  var daysPerEarthYear = 365.25;

  this.seconds = seconds;
  this.secondsPerEarthYear =
    secondsPerMinute * minutesPerHour * hoursPerDay * daysPerEarthYear;

  this.earthAge = this.seconds / this.secondsPerEarthYear;

  this.period = {
    earth: 1,
    mercury: 0.2408467,
    venus: 0.61519726,
    mars: 1.8808158,
    jupiter: 11.862615,
    saturn: 29.447498,
    uranus: 84.016846,
    neptune: 164.79132
  };
}

Age.prototype.on = function on(location) {
  return Number((this.earthAge * (1 / this.period[location])).toFixed(2));
};

Age.prototype.onEarth = function onEarth() {
  return this.on('earth');
};

Age.prototype.onMercury = function onEarth() {
  return this.on('mercury');
};

Age.prototype.onVenus = function onEarth() {
  return this.on('venus');
};

Age.prototype.onMars = function onEarth() {
  return this.on('mars');
};

Age.prototype.onJupiter = function onEarth() {
  return this.on('jupiter');
};

Age.prototype.onSaturn = function onEarth() {
  return this.on('saturn');
};

Age.prototype.onUranus = function onEarth() {
  return this.on('uranus');
};

Age.prototype.onNeptune = function onEarth() {
  return this.on('neptune');
};

module.exports = Age;
